app/Makefile

```diff
@@ -19,5 +21,10 @@ dev-build-server:
 	cd ./server; \
 	npm run build:dev
 
+test-ui:
+	cd ./ui; \
+	npm test
+
 .PHONY: \
 	run \
+	test \
```

Comment:
Bunch of goals missing from PHONY here. I'd probably use files rather than PHONY goals, that's the normal way to use make I think.

Is there a reason why we're using half make and half npm scripts?